src/elf/section_header.rs

```diff
@@ -442,13 +442,20 @@ if_alloc! {
             self.sh_addr as usize..(self.sh_addr as usize).saturating_add(self.sh_size as usize)
         }
         /// Parse `count` section headers from `bytes` at `offset`, using the given `ctx`
+        /// Assuming this is read from the whole file, it will check offset.
         #[cfg(feature = "endian_fd")]
-        pub fn parse(bytes: &[u8], mut offset: usize, mut count: usize, ctx: Ctx) -> error::Result<Vec<SectionHeader>> {
-            use scroll::Pread;
+        pub fn parse(bytes: &[u8], offset: usize, count: usize, ctx: Ctx) -> error::Result<Vec<SectionHeader>> {
```

Comment:
Build was complaining with mutability of `offset` and `count`, but I am unsure whether this is a change we want from an API point of view. Will remove if you think we should.

Reply:
yes, I get the meaning of the change, I was unsure about how it could impact your API :)